diff --git a/Porting/Maintainers.pl b/Porting/Maintainers.pl
index bf47b40..f886c57 100755
--- a/Porting/Maintainers.pl
+++ b/Porting/Maintainers.pl
@@ -1378,7 +1378,7 @@ use File::Glob qw(:case);
     },
 
     'Time::HiRes' => {
-        'DISTRIBUTION' => 'RJBS/Time-HiRes-1.9728.tar.gz',
+        'DISTRIBUTION' => 'RJBS/Time-HiRes-1.9741.tar.gz',
         'FILES'        => q[dist/Time-HiRes],
     },
 
