From 1cfa8a0c107fb5aaa0c951592265b137274143b9 Mon Sep 17 00:00:00 2001
From: Karl Williamson <khw@cpan.org>
Date: Fri, 3 May 2019 12:13:51 -0600
Subject: [PATCH] PATCH: [perl #128972] Build failure on OS X

---
 hints/darwin.sh | 14 ++++++++++++++
 1 file changed, 14 insertions(+)

diff --git a/hints/darwin.sh b/hints/darwin.sh
index feb15f6b4f..53650d26cb 100644
--- a/hints/darwin.sh
+++ b/hints/darwin.sh
@@ -273,19 +273,24 @@ add_macosx_version_min () {
   esac
 }
 
+time_hires_early_ccflags="-Werror=partial-availability -D_DARWIN_FEATURE_CLOCK_GETTIME=0"
+
 # Perl bundles do not expect two-level namespace, added in Darwin 1.4.
 # But starting from perl 5.8.1/Darwin 7 the default is the two-level.
 case "$osvers" in  # Note: osvers is the kernel version, not the 10.x
 1.[0-3].*) # OS X 10.0.x
    lddlflags="${ldflags} -bundle -undefined suppress"
+   ccflags="$ccflags $time_hires_early_ccflags"
    ;;
 1.*)       # OS X 10.1
    ldflags="${ldflags} -flat_namespace"
    lddlflags="${ldflags} -bundle -undefined suppress"
+   ccflags="$ccflags $time_hires_early_ccflags"
    ;;
 [2-6].*)   # OS X 10.1.x - 10.2.x (though [2-4] never existed publicly)
    ldflags="${ldflags} -flat_namespace"
    lddlflags="${ldflags} -bundle -undefined suppress"
+   ccflags="$ccflags $time_hires_early_ccflags"
    ;;
 [7-9].*)   # OS X 10.3.x - 10.5.x
    lddlflags="${ldflags} -bundle -undefined dynamic_lookup"
@@ -293,6 +298,7 @@ case "$osvers" in  # Note: osvers is the kernel version, not the 10.x
        *MACOSX_DEPLOYMENT_TARGET*) ;;
        *) ld="env MACOSX_DEPLOYMENT_TARGET=10.3 ${ld}" ;;
    esac
+   ccflags="$ccflags $time_hires_early_ccflags"
    ;;
 *)        # OS X 10.6.x - current
    # The MACOSX_DEPLOYMENT_TARGET is not needed,
@@ -304,6 +310,14 @@ case "$osvers" in  # Note: osvers is the kernel version, not the 10.x
     10.*)
       add_macosx_version_min ccflags $MACOSX_DEPLOYMENT_TARGET
       add_macosx_version_min ldflags $MACOSX_DEPLOYMENT_TARGET
+
+      case "$MACOSX_DEPLOYMENT_TARGET" in
+        10.[6-9]|10.10|10.11)
+        ccflags="$ccflags $time_hires_early_ccflags"
+        ;;
+      *)
+        ;;
+      esac
       ;;
     '')
       # Empty MACOSX_DEPLOYMENT_TARGET is okay.
-- 
2.17.1

